src/pgwire/protocol.rs

```diff
@@ -504,13 +504,16 @@ where
         };
 
         let param_types = stmt.param_types();
-        if param_types.len() > raw_params.len() {
+        if param_types.len() != raw_params.len() {
```

Comment:
I haven't actually tested PG's behaviour here, but providing more than required should be fine, shouldn't it? I'm imagining a scenario where someone is mechanically constructing a query that winds up, by whatever means, omitting a reference to a given placeholder. It shouldn't cease to be valid to send the entire set of placeholders in that case.

Reply:
Possibly this is true, but real Postgres rejects queries with too many parameters. I developed the test suite against real Postgres by accident (forgot to override the default connection params), and then just fixed up materialized to match. Probably best to err on the more restrictive side, since loosening restrictions will be backwards compatible but the reverse is not?

Reply:
Yeah totally, if that's what Postgres does then no objections!